quantum/matrix.c

```diff
@@ -25,37 +25,65 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include "debug.h"
```

Comment:
This file got a significant rewrite to clean up the matrix code. It is now more efficient when scanning row2col. IT NEEDS TO BE TESTED ON MULTIPLE KEYBOARDS though! I only tested it on three.